src/axom/core/tests/core_benchmark_array.cpp

```diff
@@ -0,0 +1,528 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "benchmark/benchmark.h"
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+
+#include "axom/fmt.hpp"
+#include "axom/CLI11.hpp"
+
+#include <tuple>
+#include <vector>
+#include <type_traits>
+#include <cstdint>
+
+namespace
+{
+
+enum class ArrayFeatureBenchmarks
+{
+  None = 0,
+  Constructors = 1 << 0,
+  Insertion = 1 << 1,
+  Iterators = 1 << 2,
+
+  All = Constructors | Insertion | Iterators
+};
+
+inline ArrayFeatureBenchmarks operator|(ArrayFeatureBenchmarks lhs,
+                                        ArrayFeatureBenchmarks rhs)
+{
+  using T = std::underlying_type_t<ArrayFeatureBenchmarks>;
+  return static_cast<ArrayFeatureBenchmarks>(static_cast<T>(lhs) |
+                                             static_cast<T>(rhs));
+}
+
+inline ArrayFeatureBenchmarks& operator|=(ArrayFeatureBenchmarks& lhs,
+                                          ArrayFeatureBenchmarks rhs)
+{
+  lhs = lhs | rhs;
+  return lhs;
+}
+
+inline ArrayFeatureBenchmarks operator&(ArrayFeatureBenchmarks lhs,
+                                        ArrayFeatureBenchmarks rhs)
+{
+  using T = std::underlying_type_t<ArrayFeatureBenchmarks>;
+  return static_cast<ArrayFeatureBenchmarks>(static_cast<T>(lhs) &
+                                             static_cast<T>(rhs));
+}
+
+std::vector<int> args_benchmark_sizes;
+ArrayFeatureBenchmarks args_benchmark_features {ArrayFeatureBenchmarks::None};
+}  // namespace
+
+// Custom fmt formatter for ArrayFeatureBenchmarks
+template <>
+struct axom::fmt::formatter<ArrayFeatureBenchmarks>
```

Comment:
I'm surprised fmt requires this much code to print an enum.

Reply:
Agreed -- it's because I'm using the enum for flags and wanted to print all enabled features, e.g.
```
>./tests/core_benchmark_array -f constructors insertion
[INFO] Parsed and processed command line arguments: 
[INFO] - Array sizes: 65536 
[INFO] - Array features to test: Constructors|Insertion   # <-- Output for two flags

...
>./tests/core_benchmark_array -f constructors insertion -f all
[INFO] Parsed and processed command line arguments: 
[INFO] - Array sizes: 65536 
[INFO] - Array features to test: All                      # <-- Output for "all" flags
```
 Happy to improve it if you know of a better way.